apis/python/src/tiledbsoma/_query_condition.py

```diff
@@ -136,8 +136,9 @@ def init_query_condition(
         config: Optional[dict],
         timestamps: Optional[Tuple[OpenTimestamp, OpenTimestamp]],
     ):
+        ctx = tiledb.Ctx(config)
         qctree = QueryConditionTree(
-            tiledb.open(uri, ctx=tiledb.Ctx(config), timestamp=timestamps), query_attrs
+            ctx, tiledb.open(uri, ctx=ctx, timestamp=timestamps), query_attrs
```

Comment:
@nguyenv @thetorpedodog the `ctx` is essential for the call chain

However this seems to be a back-pedal on our progress with #1169 and in particular #1664

I note that `init_query_condition` is called only from here
https://github.com/single-cell-data/TileDB-SOMA/blob/1.5.0rc1/apis/python/src/tiledbsoma/pytiledbsoma.cc#L160
where there is no user-specified context available (this is called at module-load time)

Thoughts?

Reply:
The `QueryCondition` binding code is in dire need of rewriting. It utilizes an old method we were using with Pybind11 where we did not bind from core directly but have this unnecessary `PyQueryCondition` class as an inbetween that gets initialized with `init_query_condition`. We now bind everything directly but have not gone back yet to update `QueryCondition`.

Reply:
@thetorpedodog `init_query_condition` is called here
https://github.com/single-cell-data/TileDB-SOMA/blob/1.5.0rc1/apis/python/src/tiledbsoma/pytiledbsoma.cc#L160
and implemented here
https://github.com/single-cell-data/TileDB-SOMA/blob/1.5.0rc1/apis/python/src/tiledbsoma/query_condition.cc#L140

@thetorpedodog I don't truly understand `init_query_condition` either (yet, anyway)

@nguyenv in terms of merging this PR -- which as-is will cause a thread-count regression, defeating some work on #1664 / #1169 -- can we set up a call to develop a plan to design and accomplish the necessary refactor within a 1.5.0 timeframe?

Reply:
The API had two `init()` calls for `QueryCondition()` in lines 96 to 169 (if you include comments) of [tiledb/sm/cpp_api/query_condition.h](https://github.com/TileDB-Inc/TileDB/blob/d497fffbaba54cf66eb366f696fc7b132b9954f5/tiledb/sm/cpp_api/query_condition.h#L96-L169), one of these take a `ctx`.

Reply:
> in terms of merging this PR -- which as-is will cause a thread-count regression, defeating some work on https://github.com/single-cell-data/TileDB-SOMA/pull/1664 / https://github.com/single-cell-data/TileDB-SOMA/issues/1169 -- can we set up a call to develop a plan to design and accomplish the necessary refactor within a 1.5.0 timeframe?

Yes we can discuss this tomorrow. I do not think that updating the `QueryCondition` bindings will take more than one days worth of work.

Reply:
#1775 has been merged in, resolving these issues